internal/security/bootstrapper/command/setupacl/command.go

```diff
@@ -188,6 +204,110 @@ func (c *cmd) GetCommandName() string {
+	}
+
+	for _, roleName := range roleNames {
+		// create roles based on the service keys as the role names
+		// in phase 2, we are using the same policy rule for all services
+		edgexACLTokenRole := NewRegistryRole(roleName, ClientType, []Policy{
+			rolePolicy,
+		}, false)
+
+		// fail all if any one of the role creation failed
+		if err := c.createRole(secretstoreToken, edgexACLTokenRole); err != nil {
+			return fmt.Errorf("failed to create edgex role: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (c *cmd) createAndSaveManagementToken(bootstrapACLTokenID string) error {
+	mgmtPolicyName := "edgex-management-policy"
+	mgmtPolciyRules := `
+	# EdgeX management policy for management token in HCL format
+	acl = "read"
+	agent_prefix "" {
+		policy = "write"
+	}
+	key_prefix "" {
+		policy = "write"
+	}
+	node_prefix "" {
+		policy = "write"
+	}
+	operator = "write"
+	service_prefix "" {
+		policy = "write"
+	}
+	session_prefix "" {
+		policy = "write"
+	}
+	`
+
+	edgexMgmtPolicy, err := c.createRegistryPolicy(bootstrapACLTokenID, mgmtPolicyName, mgmtPolciyRules)
+	if err != nil {
+		return fmt.Errorf("failed to create edgex management policy: %v", err)
+	}
+
+	unlimitedDuration := "0s"
+	createMgmtToken := NewCreateRegistryToken("EdgeX registry management token", []Policy{
+		*edgexMgmtPolicy,
+	}, true, &unlimitedDuration)
+	mgmtTokenInfo, err := c.createNewToken(bootstrapACLTokenID, createMgmtToken)
+	if err != nil {
+		return err
+	}
+
+	if err := c.saveManagementToken(mgmtTokenInfo); err != nil {
+		return err
+	}
+
+	c.loggingClient.Infof("successfully created and saved registry management token with policy name %s", mgmtPolicyName)
+
+	return nil
+}
+
+func (c *cmd) saveManagementToken(tokenInfo string) error {
+	// Write the token to the specified file
+	tokenFileAbsPath, err := filepath.Abs(c.configuration.StageGate.Registry.ACL.ManagementTokenPath)
+	if err != nil {
+		return fmt.Errorf("failed to convert mgmt tokenFile to absolute path %s: %s",
+			c.configuration.StageGate.Registry.ACL.ManagementTokenPath, err.Error())
+	}
+
+	// create the directory of tokenfile if not exists yet
+	dirOfToken := filepath.Dir(tokenFileAbsPath)
+	fileIoPerformer := fileioperformer.NewDefaultFileIoPerformer()
+	if err := fileIoPerformer.MkdirAll(dirOfToken, 0700); err != nil {
+		return fmt.Errorf("failed to create tokenpath base dir: %s", err.Error())
+	}
+
+	fileWriter, err := fileIoPerformer.OpenFileWriter(tokenFileAbsPath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600)
+	if err != nil {
+		return fmt.Errorf("failed to open file writer %s: %s", tokenFileAbsPath, err.Error())
+	}
+
+	if _, err := fileWriter.Write([]byte(tokenInfo)); err != nil {
+		_ = fileWriter.Close()
+		return fmt.Errorf("failed to write tokenInfo: %s", err.Error())
+	}
+
+	if err := fileWriter.Close(); err != nil {
+		return fmt.Errorf("failed to close token file: %s", err.Error())
+	}
+
+	c.loggingClient.Infof("token is written to %s", tokenFileAbsPath)
+
+	return nil
+}
```

Comment:
This code seems repeated. Similar code to saving sentinel file or other tokens? If so should create helper function that does this in a generic way.

Reply:
nope, this one is special version for management token alone with configuration and only save the raw json string not the json.encoder for this one.

Reply:
ok